src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSupport.java

```diff
@@ -93,7 +119,7 @@ protected CompletableFuture<List<LSPUsagePsiElement>> doLoad(LSPUsageSupportPara
                         }
 
                         // Collect definitions
-                        if (clientFeature.getDefinitionFeature().isDefinitionSupported(file)) {
+                        if (collectDefinitions && clientFeature.getDefinitionFeature().isDefinitionSupported(file)) {

```

Comment:
And now we conditionally search for definitions or type definitions if we definitively determined that the searched identifier is a type or is not a type above. If we couldn't determine, we'll search for both just as before.